tutorials/20_REST_API.ipynb

```diff
@@ -0,0 +1,538 @@
+    "* **REST API** allows other applications to interact with Haystack by handling their queries and returning responses.\n",
+    "* **Docker** simplifies the environment setup needed to have Elasticsearch running.\n",
+    "\n",
+    "This tutorial introduces you to all the concepts needed to build an end-to-end document search application. After completing this tutorial, you will have learned how to create a pipeline YAML file, index files, and how to query your application using REST API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preparing the Environment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Update or install Docker and Docker Compose, and launch Docker."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "apt-get update && apt-get install docker && apt-get install docker-compose\n",
+    "service docker start"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. Install Haystack.\n",
+    "\n",
+    "Install the latest version of Haystack from the main branch and all its dependencies. REST API and `xpdf` are also required.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash \n",
+    "\n",
+    "pip install --upgrade pip\n",
+    "pip install 'farm-haystack[all]'\n",
+    "pip install -e rest_api/\n",
+    "\n",
+    "brew install xpdf # required for PDFToTextConverter node"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "3. Clone Haystack repository.\n",
+    "\n",
+    "Haystack provides a `docker-compose.yml` file that defines a container for Elasticsearch. Clone the Haystack repository to be able to run the `docker-compose.yml` file locally. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "git clone https://github.com/deepset-ai/haystack.git"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "4. Update the `docker-compose.yml` file.\n",
+    "\n",
+    "The current `docker-compose.yml` file has basic configs and it also provides an image for Elasticsearch container which needs to change. For that, go to `docker-compose.yml` and replace `image: \"deepset/elasticsearch-countries-and-capitals\"` with `image: \"docker.elastic.co/elasticsearch/elasticsearch:7.9.2\"`. The new image is going to provide an empty Elasticsearch instance instead of an instance with some indexed articles about countries and capital cities. "
```

Comment:
This feels a bit confusing, modifying a compose file that's meant for another app. 

Also if we only run one container (ES) there's no need for a compose file at all, you could just `docker run` Elasticsearch. So if we want to use a compose file, I would also use the Haystack image already shipping `rest_api` and `xpdf`, that would simplify the overall setup. Also I would use a dedicated compose file users can copypaste without modification.

And if we go